content/posts/2018-01-10-kotlin-null.md

```diff
@@ -0,0 +1,249 @@
+---
+title: "Kotlin: The Problem with `null`"
+date: 2018-01-10
+slug: kotlin-the-problem-with-null
+---
+
+I think almost everybody in the CS world knows about the [Null References: The Billion Dollar Mistake](https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare) talk.
+However, it does not really matter if a concept of `null` references a good idea or a really bad one.
+We live in the world where we just have to deal with `null` this way or another.
+Yep, just deal with it :sunglasses:. So let’s ask the “How” question instead of the “Why” one.
+
+# Enter Kotlin
+
+The drug of choice of mine at this point in time is Kotlin,
+but you cannot forget about Java when using Kotlin.
+Well, the JVM probably can be avoided by using Kotlin Native,
+but at least on Android you are stuck with the JVM as an intermediate bytecode provider.
+
+Kotlin was, and is a great language.
+Backed by a team at JetBrains behind the best IDE platform available,
+reasonable and pragmatic design decisions — what else to wish for?
+And, more importantly, it brought the concept of Null Safety
+to the production table for developers.
+
+All Java objects are essentially implicitly can be `null`,
+when Kotlin objects have to be declared nullable explicitly,
+otherwise the type system assumes them be to be non-nullable by default.
+
+Finally! We can use the JVM and forget about the `NullPointerException`!
+[Well, not exactly](https://en.wikipedia.org/wiki/Forgot_About_Dre).
+
+Let’s take a look at a very short Kotlin + RxJava sample.
+
+```kotlin
+Observable.just(null).subscribe()
```

Comment:
I would rather stick with one concrete example through the whole blog post.  Either this (`Observable.just()`) or  the `CorporateCalculator`. Since you are going to mentioned the facade pattern afterwards and I guess it's much more work to do build a facade around RxJava I would recommend to use the `CorporateCalculator` as the example in this blog post

Reply:
Yep, I thought about it for a while and declined the motion. Mostly because this part points out that there are compatibility issues even in popular and modern libraries. The `Calculator` is a pure sample detached from any meaning. But I feel you ;-)

Reply:
👍